### DISCUSSION TITLE
网站启动sw.js后评论无法正常加载

### DISCUSSION THREAD

**Discussion by imjason** [walinejs/waline#351] [Q&A] [answered]
正常显示连接为：qystudio.ltd/links/
启动sw加速后：qystudio.ltd/links/index.html?_sw-precache=1d9f0bff9ac1dea33a14d2486db1f9b7

**Comment by lizheming** [comment]
![image](https://user-images.githubusercontent.com/424491/122569525-69d52d80-d07d-11eb-9ee6-30f3611d6841.png)
我试了下是可以的？

**Comment by Mister-Hope** [comment]
官网的评论也是有sw的，我不觉得sw会出问题，因为我们唯一读url的地方就是设置的path，而默认的location.path是不含url参数的。

建议给个可以复现的流程或者录屏。

**Comment by imjason** [comment]
我指的是sw.js的页面和非sw的被识别为不同页面，原有的(非sw)的不能显示在sw的页面上

**Reply by lizheming** [reply]
默认使用的是 url 的 pathname 来作为评论的 parent id 的，你这两个地址一个是 `/links/` 一个是 `/links/index.html` 所以会不一样。如果是这种情况建议你自己在 Waline 的配置里自行设置一个唯一 id，比如 `path: 'links'` 即可。

**Comment by Mister-Hope** [comment, marked as answer]
如果你的网站没有把  `/xxx/index.html` 重定向到 `/xxx/` 的行为，你可以传入以下的 path:

```js
window.location.pathname.replace(/\/index\.html$/, '/').replace(/\.html$/, '')
```

总之，你可以自行确保显示为同一页面的不同路径最终被转换到相同的path。
